### PR TITLE
pydantic2 migration made integration tests green

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Bonus:
- fixed issues with missing common-library across multiple tests and services
- disabled openentelemtry when putting up simcore stack for tests

## Related issue/s

- requires https://github.com/ITISFoundation/osparc-services/pull/157

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by giancarloromeo — APPROVED** [review]

File: packages/pytest-simcore/src/pytest_simcore/docker_compose.py [hunk 1]
```diff
@@ -118,6 +118,10 @@ def env_file_for_testing(
             file=fh,
         )
         for key, value in sorted(testing_environ_vars.items()):
+            # ensure OpenTelemetry is not enabled
+            if key.startswith("TRACING_") or key.endswith("_TRACING"):
```
Comment (pcrespov) [inline packages/pytest-simcore/src/pytest_simcore/docker_compose.py:122]: IMO this is calling for troubles ... 
I would rather specify that explicitly somewhere and not here hidden
Why not simply specify that in the main `conftest.py` of every test?

File: packages/pytest-simcore/src/pytest_simcore/simcore_storage_service.py [hunk 2]
```diff
@@ -45,22 +45,24 @@ async def storage_service(
...
+            scheme=url_obj.scheme,
             host=storage_endpoint.host,
-            port=f"{storage_endpoint.port}",
-            path=url.path,
-            query=url.query,
+            port=storage_endpoint.port,
+            path=url_obj.path.lstrip("/"),
+            query=url_obj.query,
         )
+        return f"{storage_endpoint_url}"
```
Comment (sanderegg) [inline packages/pytest-simcore/src/pytest_simcore/simcore_storage_service.py:60]: any option to use URL from yarl and do something to change the port? at this point this looks very cumbersome.

File: services/dynamic-sidecar/requirements/prod.txt [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
 # installs this repo's packages
 simcore-models-library @ ../../packages/models-library/
+simcore-common-library @ ../../packages/common-library/
```
Comment (sanderegg) [inline services/dynamic-sidecar/requirements/prod.txt:14]: same her

File: services/storage/src/simcore_service_storage/models.py [hunk 4]
```diff
@@ -191,7 +191,7 @@ def convert_from_lower_case(cls, v: str) -> str:
     def when_directory_force_link_type_and_file_size(cls, data: Any) -> Any:
         assert isinstance(data, dict)
 
-        if bool(data.get("is_directory", False)) is True:
+        if TypeAdapter(bool).validate_python(data.get("is_directory", "false")):
```
Comment (sanderegg) [inline services/storage/src/simcore_service_storage/models.py:194]: "false" vs "False" ?

File: services/web/server/src/simcore_service_webserver/exporter/_formatter/xlsx/code_description.py [hunk 5]
```diff
@@ -28,8 +28,8 @@ class RRIDEntry(BaseModel):
 
 class TSREntry(BaseModel):
     references: list[str]
-    target_level: int | None  # max value allowed
-    current_level: int | None  # current selection
+    target_level: int | None = None  # max value allowed
+    current_level: int | None = None  # current selection
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/exporter/_formatter/xlsx/code_description.py:32]: sure about that change?

**Review by sanderegg — COMMENTED** [review]
a few things

File: packages/pytest-simcore/src/pytest_simcore/docker_compose.py [hunk 6]
```diff
@@ -94,13 +94,37 @@ def testing_environ_vars(env_devel_file: Path) -> EnvVarsDict:
...
+            "INVITATIONS_TRACING",
+            "PAYMENTS_TRACING",
+            "RESOURCE_USAGE_TRACKER_TRACING",
+            "STORAGE_TRACING",
+            "WB_DB_EL_TRACING",
+            "WB_GC_TRACING",
+            "WEBSERVER_TRACING",
+        )
+    }
+
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/pytest-simcore/src/pytest_simcore/docker_compose.py:120]: yeah until people start testing tracing.

File: services/storage/src/simcore_service_storage/models.py (hunk 4 above)
Comment (sanderegg) [reply services/storage/src/simcore_service_storage/models.py:194]: so why the change?

File: packages/models-library/src/models_library/api_schemas_catalog/services.py [hunk 7]
```diff
@@ -205,7 +205,10 @@ class ServiceUpdate(ServiceMetaDataEditable, ServiceAccessRights):
 class ServiceGet(
     ServiceMetaDataPublished, ServiceAccessRights, ServiceMetaDataEditable
 ):  # pylint: disable=too-many-ancestors
-    owner: LowerCaseEmailStr | None
+    owner: LowerCaseEmailStr | None = Field(
+        None,
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_catalog/services.py:209]: this change seems wrong. the ServiceGet comes from the catalog. this changes the API are you sure?

File: packages/pytest-simcore/src/pytest_simcore/docker_registry.py [hunk 8]
```diff
@@ -261,7 +261,7 @@ def jupyter_service(docker_registry: str, node_meta_schema: dict) -> dict[str, A
     )
 
 
-@pytest.fixture(scope="session", params=["2.0.4"])
+@pytest.fixture(scope="session", params=["2.0.7"])
```
Comment (sanderegg) [inline packages/pytest-simcore/src/pytest_simcore/docker_registry.py:264]: question: should this not go into upstream master?

File: services/storage/src/simcore_service_storage/models.py (hunk 4 above)
Comment (GitHK) [reply services/storage/src/simcore_service_storage/models.py:194]: Because of the following:

```python
>>> data = {"is_directory": "false"}
>>> bool(data.get("is_directory", False))
True
```

File: packages/pytest-simcore/src/pytest_simcore/helpers/storage.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+from collections.abc import Callable
+
+from yarl import URL
+
+
+def replace_storage_endpoint(host: str, port: int) -> Callable[[str], str]:
+    def _(url: str) -> str:
+        url_obj = URL(url).with_host(host).with_port(port)
```
Comment (sanderegg) [inline packages/pytest-simcore/src/pytest_simcore/helpers/storage.py:8]: thanks!

File: services/agent/requirements/prod.txt [hunk 10]
```diff
@@ -11,6 +11,7 @@
 
 # installs this repo's packages
 simcore-models-library @ ../../packages/models-library
+simcore-common-library @ ../../packages/common-library/
```
Comment (sanderegg) [inline services/agent/requirements/prod.txt:14]: I think you forgot all the ci.txt and devel.txt files.

File: services/web/server/tests/integration/01/test_garbage_collection.py [hunk 11]
```diff
@@ -344,9 +345,16 @@ async def disconnect_user_from_socketio(
...
+
+    async for attempt in AsyncRetrying(
+        wait=wait_fixed(0.1), stop=stop_after_delay(5), reraise=True
+    ):
+        with attempt:
+            assert not await socket_registry.find_keys(("socket_id", sio.get_sid()))
+            assert sid not in await socket_registry.find_resources(
+                resource_key, "socket_id"
+            )
+            assert not await socket_registry.find_resources(resource_key, "socket_id")
```
Comment (sanderegg) [inline services/web/server/tests/integration/01/test_garbage_collection.py:357]: what does this have to do with pydantic v2?
if this is a fix for master this should go there. otherwise merging changes will become a nightmare.

**Review by sanderegg — COMMENTED** [review]
Please check my comments, there are a few left and I think you forgot all the dev.txt and ci.txt files

File: services/storage/src/simcore_service_storage/models.py (hunk 4 above)
Comment (sanderegg) [reply services/storage/src/simcore_service_storage/models.py:194]: ok all clear

File: packages/pytest-simcore/src/pytest_simcore/docker_registry.py (hunk 8 above)
Comment (GitHK) [reply packages/pytest-simcore/src/pytest_simcore/docker_registry.py:264]: no we don't use it there for anything. It's OK like this, since this is used for the CI.

File: services/web/server/tests/integration/01/test_garbage_collection.py (hunk 11 above)
Comment (GitHK) [reply services/web/server/tests/integration/01/test_garbage_collection.py:357]: OK, reverting. Did not think about that.

File: services/agent/requirements/prod.txt (hunk 10 above)
Comment (GitHK) [reply services/agent/requirements/prod.txt:14]: checked all of them, should be OK now

File: packages/models-library/src/models_library/api_schemas_catalog/services.py (hunk 7 above)
Comment (GitHK) [reply packages/models-library/src/models_library/api_schemas_catalog/services.py:209]: I know we have the following scenario:
-> push service to osparc registry who's email is not part of the users
-> this owner filed is set to None

This should cover this case where. Since an integration test was failing this helped with fixing it. I'm not 100% sure if it might break anything though.

Comment (sanderegg) [reply packages/models-library/src/models_library/api_schemas_catalog/services.py:209]: this is the API of the catalog so I guess this should not be changed

**Review by pcrespov — APPROVED** [review]

File: packages/pytest-simcore/src/pytest_simcore/docker_compose.py [hunk 12]
```diff
@@ -94,13 +94,37 @@ def testing_environ_vars(env_devel_file: Path) -> EnvVarsDict:
     # ensure we do not use the bucket of simcore or so
     env_devel["S3_BUCKET_NAME"] = "pytestbucket"
 
+    # ensure OpenTelemetry is not enabled
+    env_devel |= {
+        x: "null"
+        for x in (
+            "AGENT_TRACING",
```
[the remark this answers is not in the document]
Comment (GitHK) [reply packages/pytest-simcore/src/pytest_simcore/docker_compose.py:101]: Please note that these settings are only used inside `simcore_docker_compose` fixture. My initial approach was to set all of the settings ending in`_TRACING` to `null`.
The post against this approach was: `How do I figure out it's disabled?` -> pretty hard since you cannot search for `SERVICE_TRACING`  and figure out where its used.
Since the scope is pretty limited to only osparc services that are used for tests, I think this might be the better approach.
If something bothers you you specifically disable it.


The other approaches where every single service has testing disabled in contests was already applied to everywhere. So we should be OK like this.

**Review by sanderegg — APPROVED** [review]
thanks!

File: packages/models-library/src/models_library/api_schemas_catalog/services.py [hunk 13]
```diff
@@ -205,10 +205,7 @@ class ServiceUpdate(ServiceMetaDataEditable, ServiceAccessRights):
 class ServiceGet(
     ServiceMetaDataPublished, ServiceAccessRights, ServiceMetaDataEditable
 ):  # pylint: disable=too-many-ancestors
-    owner: LowerCaseEmailStr | None = Field(
-        None,
-        description="None when the owner email cannot be found in the database",
```
Comment (pcrespov) [inline packages/models-library/src/models_library/api_schemas_catalog/services.py]: MINOR: keep the description or as comment. I think it is informative of why None is used for
